src/Package/Impl/RClient/MicrosoftRClient.cs

```diff
@@ -0,0 +1,83 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.IO;
+using Microsoft.Common.Core;
+using Microsoft.Common.Core.OS;
+using Microsoft.Common.Core.Shell;
+using Microsoft.R.Support.Settings;
+using Microsoft.VisualStudio.R.Package.Options.R;
+using Microsoft.VisualStudio.R.Packages.R;
+using Microsoft.Win32;
+
+namespace Microsoft.VisualStudio.R.Package.RClient {
+    internal static class MicrosoftRClient {
```

Comment:
Good question. This is called on UI thread (at package load) and should be. Add assert?